sphinx_needs/filter_common.py

```diff
@@ -306,9 +306,14 @@ def filter_single_need(
         # Set filter_context as globals and not only locals in eval()!
         # Otherwise, the vars not be accessed in list comprehensions.
         if filter_compiled:
-            result = bool(eval(filter_compiled, filter_context))
+            result = eval(filter_compiled, filter_context)
         else:
-            result = bool(eval(filter_string, filter_context))
+            result = eval(filter_string, filter_context)
+        if isinstance(result, str):
```

Comment:
It makes more sense to check here, if the result is **not** a boolean value, as this is the only allowed return type for a filter.

Reply:
I have convert result to boolean when it is returned. I see that when we use the string filter like "search("AAA", title)" -> the `result ` type is re.Match type, this result is not a boolean but it also valid
![image](https://github.com/useblocks/sphinx-needs/assets/134590104/66b36c4e-769f-483c-a73d-e91e44a85500)

Reply:
Mhh then this is a bug for `search()`. And luckily `search()` returns `None`, which gets transformed to `False`, or it returns a re.Match object, which is always `True`.
So we should fix our `search()` implementation as well and I would stay with my opinion, that we should only allow Boolean returns values for `eval()`.

Checking for `string` only solves some specific problems, but is not a generic solution, which detects all faulty filter strings.